crates/katana/rpc/rpc/tests/starknet.rs

```diff
@@ -172,5 +176,171 @@ async fn test_send_declare_and_deploy_legacy_contract() {
+    );
+
+    // Prepare Katana + Messaging Contract
+    let messagin_config = json!({
+        "chain": "ethereum",
+        "rpc_url": anvil_runner.endpoint,
+        "contract_address": contract_strk.address().to_string(),
+        "sender_address": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+        "private_key": "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
+        "interval": 2,
+        "from_block": 0
+    });
+    let serialized_json =
+        serde_json::to_string_pretty(&messagin_config).expect("Failed to serialize JSON");
+
+    // Write JSON string to a file
+    let mut file =
+        File::create("tests/test_data/anvil-messaging.json").expect("Failed to create file");
+    file.write_all(serialized_json.as_bytes()).expect("Failed to write to file");
+
+    let katana_runner =
+        KatanaRunner::new_with_messaging("tests/test_data/anvil-messaging.json".to_string())
+            .unwrap();
+    let starknet_account = katana_runner.account(0);
+
+    let path: PathBuf = PathBuf::from("tests/test_data/cairo_l1_msg_contract.json");
+    let (contract, compiled_class_hash) =
+        common::prepare_contract_declaration_params(&path).unwrap();
+
+    let class_hash = contract.class_hash();
+    let res =
+        starknet_account.declare(Arc::new(contract), compiled_class_hash).send().await.unwrap();
+
+    let receipt = TransactionWaiter::new(res.transaction_hash, starknet_account.provider())
+        .with_tx_status(TransactionFinalityStatus::AcceptedOnL2)
+        .await
+        .expect("Invalid tx receipt");
+
+    assert_eq!(receipt.finality_status(), &TransactionFinalityStatus::AcceptedOnL2);
+
+    assert!(
+        starknet_account
+            .provider()
+            .get_class(BlockId::Tag(BlockTag::Latest), class_hash)
+            .await
+            .is_ok()
+    );
+
+    let constructor_calldata = vec![];
+
+    let calldata = [
+        vec![
+            class_hash,                                     // class hash
+            FieldElement::ZERO,                             // salt
+            FieldElement::ZERO,                             // unique
+            FieldElement::from(constructor_calldata.len()), // constructor calldata len
+        ],
+        constructor_calldata.clone(),
+    ]
+    .concat();
+
+    let contract_address = get_contract_address(
+        FieldElement::ZERO,
+        res.class_hash,
+        &constructor_calldata,
+        FieldElement::ZERO,
+    );
+
+    let transaction = starknet_account
+        .execute(vec![Call {
+            calldata,
+            // devnet UDC address
+            to: FieldElement::from_hex_be(
+                "0x41a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf",
+            )
+            .unwrap(),
+            selector: get_selector_from_name("deployContract").unwrap(),
+        }])
+        .send()
+        .await
+        .unwrap();
```

Comment:
lets start using `starknet-rs`'s abstraction for UDC class to do contract deployment whenever we can.

https://github.com/xJonathanLEI/starknet-rs/blob/43b2e8e3ebbc146399191a3a7c8d7e0e82d88454/starknet-contract/src/factory.rs#L23